edx_rest_framework_extensions/auth/jwt/decoder.py

```diff
@@ -81,19 +83,20 @@ def decode_jwt_filters(token):
     return [jwt_filter.split(':') for jwt_filter in jwt_decode_handler(token)['filters']]
 
 
-def _set_token_defaults(token):
+def _set_token_defaults(token, suppress_logging=None):
     """
     Returns an updated token that includes default values for
     fields that were introduced since the token was created
     by checking its version number.
     """
-    def _verify_version(jwt_version):
+    def _verify_version(jwt_version, suppress_logging=None):
         supported_version = Version(
             settings.JWT_AUTH.get('JWT_SUPPORTED_VERSION', JwtTokenVersion.latest_supported)
         )
         if jwt_version.major > supported_version.major:
-            logger.info('Token decode failed due to unsupported JWT version number [%s]', str(jwt_version))
-            raise jwt.InvalidTokenError
+            if not suppress_logging:
+                logger.info('Token decode failed due to unsupported JWT version number [%s]', str(jwt_version))
+            raise jwt.InvalidTokenError('JWT version number [%s] is unsupported', str(jwt_version))
```

Comment:
Reviewers: FYI: I added the reason the token was invalid to the exception raised.